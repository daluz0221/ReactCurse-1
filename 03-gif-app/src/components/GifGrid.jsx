import React from 'react'
import PropTypes from 'prop-types'
import GifItem from './GifItem';
import { useFetch } from '../hooks/useFetch';
// import { getImg } from '../helpers/getImg';

const GifGrid = ({category}) => {
    

    const {data, loading} = useFetch(category);

   

    


    

    
    
    
    return (
        <>
            <h3>{category}</h3>
            {loading && <p>Cargando...</p>}
            <div className="card-grid">
        
                    {
                        data.map(img=>{
                            return <GifItem
                            key={img.id}
                            {...img}
                            
                            />
                        })
                    }
            
            </div>
        </>
    )
}

GifGrid.propTypes = {

}

export default GifGrid;
