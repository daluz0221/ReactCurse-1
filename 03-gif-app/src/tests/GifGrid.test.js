import '@testing-library/jest-dom';
import {shallow} from 'enzyme'
import GifGrid from '../components/GifGrid';

describe('Pruebas en el componente <GifGrid />', () => {
    
    const wrapper = shallow(<GifGrid />);

    beforeEach(()=>{
        const wrapper = shallow(<GifGrid />);

    });

    test('Debe crear el componente y contrastarlo con el snapshot', () => {
       
        expect(wrapper).toMatchSnapshot();

    });
    

});
