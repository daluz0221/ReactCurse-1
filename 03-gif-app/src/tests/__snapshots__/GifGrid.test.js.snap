// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe crear el componente y contrastarlo con el snapshot 1`] = `
<Fragment>
  <h3 />
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
